File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        if k == 0:
            return False
        
        seen = set()
        
        for i in range(min(len(nums), k + 1)):
            if nums[i] in seen:
                return True
            
            seen.add(nums[i])
        
        for i in range(k + 1, len(nums)):
            seen.remove(nums[i - k - 1])

            if nums[i] in seen:
                return True

            seen.add(nums[i])
        
        return False

File: Data Structures & Algorithms/missing-number/submission-2.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int len = nums.size();
        int xorNoSkip = 0;
        int xorSkip = 0;

        for (int i = 1; i <= len; i++) {
            xorNoSkip ^= i;
        }

        for (int i = 0; i < len; i++) {
            xorSkip ^= nums[i];
        }

        return xorNoSkip ^ xorSkip;
    }
};
